plugins/references.py

```diff
@@ -760,6 +781,92 @@ def run(self, edit, name=None):
             offset -= convert2ref(view, edit, _link_span, link_span[1], link_span[2])
 
 
+class MdeConvertBareLinkToMdLinkCommand(MdeTextCommand):
+    """Convert an inline link to reference."""
+
+    def is_visible(self):
+        """Return True if selection contains links"""
+        view = self.view
+        for sel in view.find_by_selector(barelink_scope_name):
+            if any(s.intersects(sel) for s in view.sel()):
+                return True
+        return False
+
+    def run(self, edit, name=None):
+        """Run command callback."""
+        # import queue
+        import threading
+        import time
+
+        thread_queue = []
+
+        url_titles = {}
+        url_redirects = {}
+
+        def getTitleFromUrlJob(link_href):
+            import urllib.request
+
+            resp = urllib.request.urlopen(link_href)
+            content_type = {a: b for a, b in resp.getheaders()}.get("Content-Type")
+            if content_type and not content_type.startswith("text"):
+                url_titles[link_href] = None
+                raise TypeError(f"Link {link_href!r} points to non-text content {content_type!r}")
```

Comment:
Python 3.3 doesn't support f-strings.

As long as ST3 is still a target for releases, we need to keep python 3.3 compatibility.